rhtconsulting/rhc-ose
# Support for 3.2

#### What does this PR do?

Updated the bash tools to support installing OSEv3.2. As OSE now runs dnsmasq on every node, the DNS server installed part of our tools need to be pulled out to be installed on a different host. This is now part of the "Utility" host that originally was used to install the env. This means that the Utility Host is kept around after install instead of having it automatically removed.

In addition, the current OSE v3.2 installation expects the install to be using DHCP (rather than static) for the hosts used for master and nodes. In order to use the correct DNS server, it looks for the DHCP4_DOMAIN_NAME_SERVERS parameter to populate dnsmasq correctly. To work around this, the tools now update /etc/sysconfig/network with the DNS server ip to get the correct name resolution. 
#### How should this be manually tested?

Run a provisioning, e.g.:

```
./osc-provision --num-nodes=2 --num-etcd=1 --ose-version=3.2 --key=<key>
```

Then verify that the installation was successful with 1 master + 2 nodes, in addition check that the router and registry comes up successfully. 

Bonus: Perform a build and verify that the environment works as expected and can be used to host apps.
#### Is there a relevant Issue open for this?

resolves #172 
#### Who would you like to review this?

/cc @sabre1041 @etsauer
